packages/graphql/src/schema/create-relationship-fields/create-relationship-fields.ts

```diff
@@ -134,60 +129,15 @@ export function createRelationshipFields({
             return;
         }
 
-        // ======== only on relationships to concrete entities:
-        withSourceWhereInputType({ relationshipAdapter, composer: schemaComposer, deprecatedDirectives });
-
-        // ======== only on relationships to concrete | unions:
-        // TODO: refactor
-        withConnectOrCreateInputType({
-            relationshipAdapter,
-            composer: schemaComposer,
-            userDefinedFieldDirectives,
-            deprecatedDirectives,
-        });
-
-        // ======== all relationships:
-        composeNode.addFields(
-            augmentObjectOrInterfaceTypeWithRelationshipField(relationshipAdapter, userDefinedFieldDirectives, subgraph)
-        );
-
-        withRelationInputType({
-            relationshipAdapter,
-            composer: schemaComposer,
-            deprecatedDirectives,
-            userDefinedFieldDirectives,
-        });
-
-        augmentCreateInputTypeWithRelationshipsInput({
-            relationshipAdapter,
-            composer: schemaComposer,
-            deprecatedDirectives,
-            userDefinedFieldDirectives,
-        });
-
-        augmentConnectInputTypeWithConnectFieldInput({
-            relationshipAdapter,
-            composer: schemaComposer,
-            deprecatedDirectives,
-        });
-
-        augmentUpdateInputTypeWithUpdateFieldInput({
-            relationshipAdapter,
-            composer: schemaComposer,
-            deprecatedDirectives,
-            userDefinedFieldDirectives,
-        });
-
-        augmentDeleteInputTypeWithDeleteFieldInput({
-            relationshipAdapter,
-            composer: schemaComposer,
-            deprecatedDirectives,
-        });
-
-        augmentDisconnectInputTypeWithDisconnectFieldInput({
-            relationshipAdapter,
-            composer: schemaComposer,
-            deprecatedDirectives,
-        });
+        if (relationshipTarget instanceof ConcreteEntityAdapter) {
+            createRelationshipConcreteFields({
```

Comment:
I think this would require a bigger refactor because these are entityAdapters not the entities themselves, and we haven't got isConcreteEntityAdapter() etc methods yet.